### ISSUE TITLE
Transform not propagated for child due to delay of inserting `Children`

### ISSUE THREAD

**Issue by jakobhellermann** [bevyengine/bevy#1807]
In the following code snippet, the `1.5` scale is not applied to the scene:

```rust
fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut scene_spawner: ResMut<SceneSpawner>,
) {
    let scene_holder = commands
        .spawn()
        .insert(GlobalTransform::default())
        .insert(Transform::from_scale(Vec3::splat(1.5)))
        .insert(Holder)
        .id();

    let scene_handle = asset_server.load("models/AlienCake/cakeBirthday.glb#Scene0");
    scene_spawner.spawn_as_child(scene_handle, scene_holder);
}
```

The `transform_propagate_system` only runs if either the parent is `Changed` or the child is `Changed`.
However, in this case the order of operations is
1. scene_holder `0v0` is spawned (and is `Changed` for one frame)
2. later the scene is spawned (e.g. 0v3)
  `0v3` is `Changed` for one frame.
  A `Parent(0v0)` component is scheduled to be added via commands.
3. `parent_update_system` is run, but the `insert Children` command is not yet processed
4. `transform_propagate_system` is run. `0v3` is still `Changed`, but since `0v0` has no children yet nothing happens.
5. Next frame. `0v0` now has `Children([0v3])`.
6. `parent_update_system` is run, nothing to be done
7. `transform_propagate_system` is run. This time there are children for `0v0`, but `0v3` is not changed anymore, so nothing happens.

The problem is "fixed" by either adding a 
```rust
fn force_transform_change(mut q: Query<&mut Transform>) {
    for mut t in q.iter_mut() { t.deref_mut(); }
}
```
or removing the `Changed` from [transform_propagate_system](https://github.com/bevyengine/bevy/blob/81b53d15d4e038261182b8d7c8f65f9a3641fd2d/crates/bevy_transform/src/transform_propagate_system.rs#L43).

It is also fixed by adding `.with_children(|_| {})` to the `scene_holder`, because then the `parent_update_system` can simply insert to the existing children which works without delay.

A simpler repro is [here](https://gist.github.com/jakobhellermann/753a55665a4a4873c3ad3054cde6592d).

**Comment by jakobhellermann** [comment]
The gltf loader could be fixed by directly applying the `Commands` so that the `Children(..)` are available for the later systems.

The repro can be fixed by using 
`commands.entity(parent).push_children(&[child])` instead of `.insert(Parent(parent))`.

**Comment by james7132** [comment]
Is this still an issue after #4197?

**Comment by nicopap** [comment]
I can confirm this is fixed with #4197, using the repro linked earlier.
